Xamarin.Forms.Core/Items/GroupableItemsView.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace Xamarin.Forms
+{
+	public class GroupableItemsView : SelectableItemsView
+	{
+		public static readonly BindableProperty IsGroupingEnabledProperty =
+			BindableProperty.Create(nameof(IsGroupingEnabled), typeof(bool), typeof(GroupableItemsView), false);
+
+		public bool IsGroupingEnabled
+		{
+			get => (bool)GetValue(IsGroupingEnabledProperty);
+			set => SetValue(IsGroupingEnabledProperty, value);
+		}
+
+		public static readonly BindableProperty GroupHeaderTemplateProperty =
+			BindableProperty.Create(nameof(GroupHeaderTemplate), typeof(DataTemplate), typeof(GroupableItemsView));
```

Comment:
`defaultValue` is not a mandatory argument, but it actually should be to avoid (subtle) bugs with value types.
```suggestion
			BindableProperty.Create(nameof(GroupHeaderTemplate), typeof(DataTemplate), typeof(GroupableItemsView), default(DataTemplate));
```